test/differentiation.jl

```diff
@@ -161,42 +161,38 @@ end
 
 @testset "Riemannian differentials" begin
     s2 = Sphere(2)
-    s2c = Manifolds.Curve(s2)
     p = [0.0, 0.0, 1.0]
     q = [1.0, 0.0, 0.0]
     c1(t) = geodesic(s2, q, p, t)
-    @test domain(s2c) === ℝ
 
     Xval = [-sqrt(2) / 2, 0.0, sqrt(2) / 2]
-    @test isapprox(s2, c1(π / 4), differential(c1, s2c, π / 4), Xval)
+    @test isapprox(s2, c1(π / 4), differential(s2, c1, π / 4), Xval)
```

Comment:
I would expect a differential to be a linear functional, i.e. for real-valued functions a co-vector? This does not necessarily contradict to this line, just for clarification.

Reply:
It was already discussed in the previous PR: https://github.com/JuliaManifolds/Manifolds.jl/pull/110#issuecomment-607219033 . BTW, differential of a curve $c$ at $t$ is indeed a linear functional, but as far as I know $\mathbb{R} -> T_{c(t)} M$, so it doesn't look like a covector.

Anyway, it's a good idea to clarify it in documentation.

Reply:
Ah yes, we already discussed that.

The differential just at a point c(t) is a covector :) not the complete differential. If w look at

$D_t c(t)$ this is a map from T_{c(t)}M to R, so $D_tc(t)[X]$ to be precise and its linear. So its a linear function on T_{c(t)}M or in other words an element from T*_{c(t)}M, a covector. And that is what you compute here.

As long as the vector you return is untyped, that's completely fine, so its really something for the docs, and if we decide to type something we should take that into account.

Reply:
> _edit_: ops. I confused the curve with a function on the manifold, the above argument is the other way around and your right it should returrn tangents here.

OK, so I'll just edit the documentation.

Reply:
Does the documentation look OK now?

Reply:
I am not sure about $df_t(t)$, I would'Ve thought its $df_t(t)$ (take the derivative with respect to the variable t and then plug in the value of t afterwards), but the rest looks very good.

Reply:
In $df_{t_1}(t_2)$,  $t_1$ is a point in $\mathbb{R}$ but $t_2$ is a vector from $T_{t_1} \mathbb{R}$, so using the same value in both places doesn't look right. Anyway, you can check it for a curve on the real plane: let's say we have f(t) = (1-t, t^2). The differential at $t_1$ is df_{t_1}(u) = (-u, 2 u t_1) and df_{t_1}(1) is exactly the right tangent vector.

Reply:
Hm, then I confuse something again in your notation. For me writing $df_t$ mint something different.

Reply:
I was trying to follow the notation from https://juliamanifolds.github.io/Manifolds.jl/stable/notation.html . How would you write it then?

Reply:
> Ah I see, I would write $D_tf$ to indicate taking the derivative with respect to t, D_tf(t_1)$ to indicate the (wholesaled) derivative at t_1 and D_tf(t_1[t_2]$ for your note. But for sure we can keep your notation, I was just confused,

Hmm, maybe we should change the primary notation for differentials.

Reply:
I see, yes, then your notation was correct wrt that list and I was mainly confused. The one problem I see, is, that sometimes it is useful to distinguish which variable one takes the differential of $d\varphi_p$ would for me indicate that. but I would still write $d\varphi_p(p_1)[X]$ (one could leave our $_p(p_1)$ and just write $_{p_1}$ if it is clear because we have only one variable) but then I would still write square brackets for the direction maybe?

Reply:
OK, makes sense.

Reply:
Could you expand a bit on the distinguishing variables part? At the moment we only define differentials of functions between two manifolds and it always is a map between the respective tangent spaces. Do you mean it as using different charts, taking the differential for only a subspace of the tangent space, or something else?

Reply:
Imagine the power (or product) manifold and you only take the differential with respect to the first or second argument. So for example having a function $f(p,q)$ and you look at $df_p(p_0,q_0)[X]$ where $X$ is from the tangent space of the p component at $(p_0,q_0)$. That's where you really want to keep the variable $p$ of the differential.

Reply:
I've changed it, does it look OK now?

Reply:
Yes, that looks quite good, for your one-variable-function I think you can leave out the $(t)$ since it can not (or no longer) be confused with the direction $[1]$, so $df_t[1]$ would be fine for me, too.